runtime/src/installed_scheduler_pool.rs

```diff
@@ -229,9 +226,11 @@ pub type InstalledSchedulerRwLock = RwLock<Option<DefaultInstalledSchedulerBox>>
 impl BankWithScheduler {
     #[cfg_attr(feature = "dev-context-only-utils", qualifiers(pub))]
     pub(crate) fn new(bank: Arc<Bank>, scheduler: Option<DefaultInstalledSchedulerBox>) -> Self {
-        if let Some(bank_in_context) = scheduler.as_ref().and_then(|scheduler| scheduler.context())
+        if let Some(bank_in_context) = scheduler
+            .as_ref()
+            .map(|scheduler| scheduler.context().bank())
         {
-            assert_eq!(bank.slot(), bank_in_context.slot());
+            assert!(Arc::ptr_eq(&bank, bank_in_context));
```

Comment:
This is a stronger check, and will guarantee context and passed bank are the same.

Think the change is good, so not oppposed, but trying to think through when would the slot check not be sufficient? These bank & schedulers are tied to bank_forks, which as far as i'm aware will only support a 1:1 relationship of slot:bank. I know there's edge cases (see previous outage with duplicate leader blocks), but would both have made it into bank forks?